### PR TITLE
Create _TZ3000_qeuvnohg_fuse_w_power_meter.json

### PR DESCRIPTION
**Pull request by cornim**
Support for Tongou Power Meter and Fuse (https://zigbee.blakadder.com/Tongou_TO-Q-SY1-JZT.html)

### CODE REVIEW

**Comment by BabaIsYou** [comment]
Fix #7011

**Comment by BabaIsYou** [comment]
It seems that Matsee Plus ATMS1602Z model could also share this DDF.
Could you please change the 5 first lines of your DDF
```
      	"schema": "devcap1.schema.json",
      	"manufacturername": "_TZ3000_qeuvnohg",
      	"modelid": "TS011F",
      	"product": "TO-Q-SY1-JZT",
      	"vendor": "TONGOU",
```
by these : 
```
      	"schema": "devcap1.schema.json",
      	"manufacturername": ["_TZ3000_qeuvnohg","_TZ3000_ky0fq4ho","_TZ3000_8bxrzyxz"],
      	"modelid": ["TS011F","TS011F","TS011F"],
      	"product": "Din smart relay with power monitoring",
```
thx !

**Comment by manup** [comment]
@cornim ping :)

**Comment by labor4** [comment]
Hi,
Thank you very much for this spec.
I installed it manually.

The API shows to be working, Phoscon GUI apparently not yet.

Question: What unit is `consumption`?
```
{
    "config": {
        "on": true,
        "reachable": true
    },
    "etag": "a4040393300873a5e0aa764fcafd36d4",
    "lastannounced": null,
    "lastseen": "2024-02-06T15:16Z",
    "manufacturername": "_TZ3000_qeuvnohg",
    "modelid": "TS011F",
    "name": "Consumption 25",
    "state": {
        "consumption": 5, 
        "lastupdated": "2024-02-06T15:12:55.026"
    },
    "type": "ZHAConsumption",
    "uniqueid": "18:7a:3e:ff:fe:47:80:33-01-0702"
}
```

**Comment by labor4** [comment]
I see a possible future problem with this device.
Since this is a mains switch, it is a category that should be handled with extra care.
**Because light bulbs easily survive trial'n'error config mistakes (on/off), while the consequences here might not.**

Especially in Phoscon GUI there is a problematic default config, 
where a trigger may "shutdown all devices, even if not in a group".

**So this makes it impossible by default, to keep this device safely away from human interaction.**

Because of this, it may even be worth considering, to introduce a breaking addition into the API, to secure the transition of such devices into a different category, example:
```
should fail?
PUT     '{"on":true}'     {{baseUrl}}/api/{{APIKEY}}/lights/$NUM/state
PUT     '{"on":false}'     {{baseUrl}}/api/{{APIKEY}}/lights/$NUM/state

should succeed?
PUT     '{"onforce":true}'     {{baseUrl}}/api/{{APIKEY}}/lights/$NUM/state
PUT     '{"onforce":false}'     {{baseUrl}}/api/{{APIKEY}}/lights/$NUM/state
```
So backwards-compatibility-wise it would fail by default.

I don't know the full concept of your API mechanics, you know better, but you get my meaning.

Extra problem:
If I reboot my router in front of my Phoscon Raspi, all Zigbee devices will shut down.
Possibly triggered by interface/dhcp reconfig. (didn't investigate yet, bug present in two separate locations)

Best regards

**Comment by labor4** [comment]
a note, thought this might help others in the meantime.

it appears to me `"type": "$TYPE_ON_OFF_OUTPUT",` is not yet reflected in the GUI.
I replaced it with `"type": "$TYPE_SMART_PLUG",` for the time being.

The temp sensors become listed separately, and are not properly named together with the "plug".
But they are linked (cleanly removed if I delete the plug).

For my own needs, mentioned above, I also deleted the block:
```
      				{
      					"name": "state/on"
      				},
```

so to only use it as a meter.

I also fiddled around by adding the attribute `access: "RO"` to it (I had no documentation, just copypaste from other appearances). 
But that didn't work.

**Comment by Smanar** [comment]
Honnestly, I think you can create a new DDF with the code from this one, the changes requested and your modification.
I think the code owner have give up.

**Comment by labor4** [comment]
@Smanar I think the dependency on the PR creator was only the secondary requests for more models.
But I was under the impression that the platform was not ready for mains types, conceptually (read: to-be-decided-by-the-repo-devs).

I'm lost where to find the spec.

- Do you support `$TYPE_ON_OFF_OUTPUT`?
- **As these are mains**: Should there be an additional precaution (as in: default to `RO` on the `state/on` attribute)

I think even if you put these under `$TYPE_SMART_PLUG`,
the mentioned `state/on` **is the big roadblocker (danger)**.

The phoscon GUI has all sorts of:
- "deactivate all in group"
- "shut everythin down" (now with selection)

So Mains should be handled with a different kind of caution.
If you can show me how to add `access: "RO"`, I'll test this and make another PR.
The GUI can postpone it and add the category/lock-feature when they want to (and keep the on/off switch disabled in the mean time).

Yes you could just delete `state/on`, but that smells alpha, doesn't it?

TLDR:
IMHO: `access: "RO"` on `state/on` must be honoured, otherwise no point in merging. Too dangerous.

What do you think?

**Comment by Smanar** [comment]
>    Do you support $TYPE_ON_OFF_OUTPUT?

Sure, and I think it's supported too in all third app.

>    As these are mains: Should there be an additional precaution (as in: default to RO on the state/on attribute)

This flag is not Read only, you can switch the device too using zigbee request ?
From z2M

> [Switch](https://www.zigbee2mqtt.io/devices/TS011F_din_smart_relay.html#switch)
> 
> The current state of this switch is in the published state under the state property (value is ON or OFF). To control this switch publish a message to topic zigbee2mqtt/FRIENDLY_NAME/set with payload {"state": "ON"}, {"state": "OFF"} or {"state": "TOGGLE"}. To read the current state of this switch publish a message to topic zigbee2mqtt/FRIENDLY_NAME/get with payload {"state": ""}

**Comment by labor4** [comment]
> > Do you support $TYPE_ON_OFF_OUTPUT?
> 
> Sure, and I think it's supported too in all third app.

doing so makes it a light bulb. 
there seems to be a device-id for mains in an xml definition, but not in the list of these global variables. So seemingly the concept of „mains“ is not yet ready. 

Maybe there are other ways to define it without these globals, I have no clue. Are there guides/specs?

> 
> > As these are mains: Should there be an additional precaution (as in: default to RO on the state/on attribute)
> 
> This flag is not Read only, you can switch the device too using zigbee request ? From z2M
> 

i saw the proper value would be „R“.
the key „access“ is defined in the generic items definitions under „state_on…“
but to no effect. state change still works. 
„access“ seems unchangeable. 
the api will not list it as ingested in the device’s ddf either. 

I don‘t understand your question. 

I dont ask how to flip the „on“ state, I‘m asking to implement the strict prevention of it, as i believe this is not implemented.

> > [Switch](https://www.zigbee2mqtt.io/devices/TS011F_din_smart_relay.html#switch)
> > The current state of this switch is in the published state under the state property (value is ON or OFF). To control this switch publish a message to topic zigbee2mqtt/FRIENDLY_NAME/set with payload {"state": "ON"}, {"state": "OFF"} or {"state": "TOGGLE"}. To read the current state of this switch publish a message to topic zigbee2mqtt/FRIENDLY_NAME/get with payload {"state": ""}

**Comment by Smanar** [comment]
>doing so makes it a light bulb.

Nope, it make a "On/Off output", it's the third app that decide to display a bulb or a plug.
Or you mean in the deconz application ?

>I dont ask how to flip the „on“ state, I‘m asking to implement the strict prevention of it, as i believe this is not implemented.

This field is generaly the image of the cluster on/off 0x0006 attribute 0x0000 and this one is working on 2 sides.
If this field is read only, it will be for exemple a state/open.
But I don't remember one device using the cluster 0x0006 and have a read only information.

**Comment by labor4** [comment]
> > doing so makes it a light bulb.
> 
> Nope, it make a "On/Off output", it's the third app that decide to display a bulb or a plug. Or you mean in the deconz application ?
> 

correct, this just sorted-in there by phoscon. 
i‘m just figuring out the minimum spec of both, so not both need to be fiddled with, at once. 

my aim here is that phoscon may do whatever it is ready to do, so dont go there, but look how to prevent a blackout by the assumed intention of deconz. hence i ask to disable a toggle for the api, while keeping the info of the state. 

> > I dont ask how to flip the „on“ state, I‘m asking to implement the strict prevention of it, as i believe this is not implemented.
> 
> This field is generaly the image of the cluster on/off 0x0006 attribute 0x0000 and this one is working on 2 sides. If this field is read only, it will be for exemple a state/open (for a windows position, the device is not able to open the window) But I don't remember one device using the cluster 0x0006 and have a read only information.

this is where i‘m fully out of my depth, sorry. 
but i think you speak about the connection from deconz to the device, while i ask if, regardless of this compatibility, deconz can be configured to deny the api to change the value, while just giving the GET request the state of this value. 

this way there is some safeguard if an app like phoscon „is just not ready yet“.

this may have been a no-issue for bulbs etc, but a mains switch basically takes out a building. that’s its purpose. maybe a third party app like phoscon doesnt expect this scenario yet, seemingly (by the nature of incremental development). 

so if i understood this well enough: do you think this state/on capability should be deleted in mains, or should there be a „readonly“ enforcement for the api, until such a time when the apps can handle it? in such a case third parties „may“ be allowed to enable write-access via api, but in doing so they also state they have prepared themselves for it. thus this RO condition via the API doesnt break backwards compatibility. 

does this make sense?

btw…
yes we could critizise the use case of a zigbee switch in a mains meter. personally i dont see much sense. but there it is.

**Comment by Smanar** [comment]
>do you think this state/on capability should be deleted in mains

Can use other than state/on as this one is always R/W.
You can to create new entry and set them Read Only, like this one https://github.com/dresden-elektronik/deconz-rest-plugin/blob/master/devices/generic/items/state_open_item.json

For all third app state/on is Read/Write, so I don't think on day it will change.
